src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -283,7 +283,29 @@ void Snippet::initSupportedPrimitiveDescriptors() {
 }
 
 void Snippet::selectOptimalPrimitiveDescriptor() {
-    selectPreferPrimitiveDescriptor(getImplPriority(), true);
+    auto fun = [](const ov::intel_cpu::MemoryDescPtr& curDesc,
+                  const ov::intel_cpu::MemoryDescPtr& parentDesc,
+                  const size_t& descInConfSize) {
+        if (curDesc->getPrecision() == parentDesc->getPrecision() && descInConfSize >= 3u) {
+            auto& curShape = curDesc->getShape();
+            auto& parentShape = parentDesc->getShape();
+            if (curShape.isStatic() && (parentShape == curShape) && parentShape.isStatic()) {
+                auto& curDims = curShape.getDims();
+                size_t num1count = 0;
+                for (size_t i = 0; i < curDims.size(); i++) {
+                    if (curDims[i] == 1u) {
+                        num1count++;
+                    }
+                }
+                // Check if curDims contains exactly one non-1 element or all 1 elements.
+                if (curDims.size() > 2 && ((num1count == curDims.size() - 1u) || num1count == curDims.size())) {
+                    return true;
```

Comment:
What about the case when we have all 3 input config descs incompatible with the parent ones. I.e. all parents are 'abcd' and our input configs are 'acbd' and we satisfy the highlighted checks. So, instead of proceeding to the next supported primitive descriptor (which could be 'abcd'), we would stop here (return true), and it will result into multiple reorders?
Just trying to make sure we don't affect any 'normal' flow (not related to this particular issue) and do not add unnecessary reorders.

Reply:
I am concerned that this fix could affect a significant number of eltwise subgraphs, since the specified conditions are rather common. We need to avoid this and reduce the scope of affected models as much as possible.
To do so, I suggest:
1) Make sure that there are exactly 3 inputs, and exactly two of them match the present condition (have one non-1/or all ones')
2) Make sure that the remaining input has a parent with blocked layout.

If these conditions are met, then the descriptors on broadcasted inputs can be skipped.

Also, @xipingyan, is it necessary to include the all ones' inputs (e.g. [1,1,1])? Can they also occur in the pattern of interest?
If not, then we shouldn't cover this case to reduce the scope of affected subgraphs.

Reply:
Updated:
    Add 3 conditions check:
    1: inputs num == 3;
    2: input 1 and input 3, have shape [1,?,1,1] and ncsp;
    3: input 2 have shape non-[1,?,1,1] and nspc;

Reply:
@IvanNovoselov @EgorDuplensky , updated, please review it again.